Posts

mdsanchez1 opened the pull request:
- Created the Post object, which is used for the social media function of the website.
- Created the database table which holds all the posts and reference foreign objects such as the user who made the post or the recipe connected to it.
- Created and tested (with Postman) the APIs to interact with the posts in the database.
- Needs to be merged and updated to work with the new recipe functionality.
#19 #21 #18 

- Added the API to update a posts message, image and/or recipe.
- Refactored likes/dislikes so that a user can only have one reaction on each post and they can remove them.
- Tested all of the APIs.

#20 #36 #41

mdsanchez1 committed:
[Implementation] Began implementation of posts object.

- Post object.
- Database initialization for Posts.
- Currently non-functional API routes for Posts.

mdsanchez1 committed:
Attempting to fix APIs

mdsanchez1 committed:
Posts are now added to the database using the API

mdsanchez1 committed:
Finished up the post APIs.

- Can add and retrieve posts from DB.
- Can delete, like, dislike posts as well.
- Needs to be merged with the new recipe functionality.

cammyj1122 commented on api/db/database.py:
```diff
@@ -22,35 +17,140 @@ def __init__(self):
+        self.cursor.execute(command_string, (username,))
+        user_data = self.cursor.fetchone()
+        if user_data:
+            return User(userId=user_data[0], username=user_data[1], password=user_data[2])
```
The user object is (username, password, id) while the database object returns id, username, password. Please confirm that after changing this the information is in the right order.

mdsanchez1 replied:
labeled parameters are used here to order doesnt matter

cammyj1122 commented on api/db/database.py:
```diff
@@ -22,35 +17,140 @@ def __init__(self):
-            userid: int = self.cursor.fetchone()
-            return userid[0]
-        except:
+            return True
```
Why was this reverted? It was helpful to return the user id so that it could be returned to the frontend.

mdsanchez1 replied:
probably from a merge conflict at some point, i didnt change this i will revert it

cammyj1122 commented on api/db/objects.py:
```diff
@@ -12,4 +16,14 @@ def to_dict(self) -> dict:
+    userId: int = Field(..., description="ID of the user who created the post")
+    message: str = Field(..., description="Content of the post")
+    image: Optional[str] = Field(default=None, description="Base64-encoded image data")
+    recipe: Optional[Recipe] = Field(default=None, description="Recipe associated with the post")
```
Right now, in the frontend it is possible for me to just send the recipe id to the backend, rather than the entire recipe object.

cammyj1122 commented on api/routes.py:
```diff
@@ -240,4 +241,102 @@ async def getUser(username: str) -> dict:
+    return posts
+
+@postRouter.put("/{post_id}/like", response_description="Like a post", status_code=200)
+async def like_post(post_id: int):
```
Should we make it possible to unlike and undislike a post?

cammyj1122 replied:
This was added, resolving comment

cammyj1122 commented on api/routes.py:
```diff
@@ -240,4 +241,102 @@ async def getUser(username: str) -> dict:
+        status_code=status.HTTP_404_NOT_FOUND,
+        detail=f"Post with ID {post_id} not found."
+    )
+
```
Could you add a route to get a list of posts associated with a specific user?

cammyj1122 commented on api/routes.py:
```diff
@@ -240,4 +241,102 @@ async def getUser(username: str) -> dict:
+async def create_post(post: Post):
+    """Creates a new post in the database."""
+    try:
+        if db.add_post(post):
```
We might want to add the date on the server side, rather than client.

cammyj1122 replied:
Wait, nevermind, it seems like the object does this.

mdsanchez1 committed:
Implementation and Testing of Posts

- Added the API to update a posts message, image and/or recipe.
- Refactored likes/dislikes so that a user can only have one reaction on each post and they can remove them.
- Tested all of the APIs.

mdsanchez1 committed:
Undoing AI message change

mdsanchez1 committed:
Changing Post to only hold RecipeID

cammyj1122 reviewed (approved):
All of my comments were resolved, this looks good.
